packages/ui/src/utils/erc20/index.tsx

```diff
@@ -78,3 +81,45 @@ export const sortERC20Tokens = (tokens: TokenInfoWithBalance[]) => {
         else return a.symbol ? -1 : b.symbol ? -1 : 0;
     });
 };
+
+export const tokenInfoWithBalanceEquals = (
+    tokenA?: TokenInfoWithBalance | null,
+    tokenB?: TokenInfoWithBalance | null
+): boolean => {
+    return !!(
+        tokenA &&
+        tokenB &&
+        tokenA.chainId === tokenB.chainId &&
+        tokenA.address.toLowerCase() === tokenB.address.toLowerCase()
+    );
+};
+
+export const cacheTokenInfoWithBalance = (
+    token: TokenInfoWithBalance
+): void => {
+    const serializableToken = { ...token };
+    const cachingKey = `imported-erc20-tokens-${serializableToken.chainId}`;
+    delete serializableToken.balance;
```

Comment:
Why do we need to delete the `balance` here?

Reply:
The cacher stores and retrieves data to/from `localStorage` using `JSON.stringify` and `JSON.parse` and `BigNumber` is not really serializable. Besides, caching the balance of the token would be pointless since we risk it being outdated when we have to show it to the user.